### PR TITLE
feat: Add USDC support

### PR DESCRIPTION
**Pull request by MilGard91**
## Context

Issue: #83 

## Changes proposed in this pull request

- Add USDC as a asset to the store
- Add caching overwrite for constants
- remove comments regarding support for the decimals, because PriceFeedOracle returns rates in asset units
- add asset object to `/quote`, `/capacity` and `/capacity/{productId}` responses
- update openapi docs

`/quote`
```json
{
  "quote": {
    "totalCoverAmountInAsset": "string",
    "annualPrice": "string",
    "premiumInNXM": "string",
    "premiumInAsset": "string",
    "poolAllocationRequests": [
      {
        "poolId": "string",
        "coverAmountInAsset": "string",
        "skip": false
      }
    ],
    "asset": { // <--- new asset field
      "id": "string",
      "symbol": "string",
      "decimals": 18
    }
  },
  "capacities": [
    {
      "poolId": "string",
      "capacity": [
        {
          "assetId": "string",
          "amount": "string",
          "asset": { // <--- new asset field
            "id": "string",
            "symbol": "string",
            "decimals": 18
          }
        }
      ]
    }
  ]
}
```

`/capacity/{productId}`
```json
{
  "productId": 0,
  "availableCapacity": [
    {
      "assetId": 0,
      "amount": "string",
      "asset": { // <--- new asset field
        "id": "string",
        "symbol": "string",
        "decimals": 18
      }
    }
  ],
  "allocatedNxm": "string",
  "minAnnualPrice": "string",
  "maxAnnualPrice": "string"
}
```

`/capacity`
```json
[
  {
    "productId": 0,
    "availableCapacity": [
      {
        "assetId": 0,
        "amount": "string",
        "asset": { // <--- new asset field
          "id": "string",
          "symbol": "string",
          "decimals": 18
        }
      }
    ],
    "allocatedNxm": "string",
    "minAnnualPrice": "string",
    "maxAnnualPrice": "string"
  }
]
```

## Test plan

- Add tests for USDC support
- Add tests for asset object response

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

**Comment by rackstar** [comment]
looks good. as discussed maybe consider adding decimal info for the asset for the 3rd party integration (oper cover, etc..)

**Comment by rackstar** [comment]
> looks good. as discussed maybe consider adding decimal info for the asset for the 3rd party integration (oper cover, etc..)

Added asset.decimal info

File: src/routes/quote.js [hunk 1]
```diff
@@ -169,6 +169,10 @@ router.get(
         premiumInNXM: quote.premiumInNXM.toString(),
         premiumInAsset: quote.premiumInAsset.toString(),
         poolAllocationRequests: quote.poolAllocationRequests,
+        asset: {
+          symbol: ASSET_ID_MAPPING[coverAsset].symbol,
+          decimals: ASSET_ID_MAPPING[coverAsset].decimals,
+        },
```
Comment (rackstar) [inline src/routes/quote.js:175]: @MilGard91 I added as well `asset.symbol` for convenience. should we keep it or drop it?

File: src/lib/constants.js [hunk 2]
```diff
@@ -1,7 +1,14 @@
 const { BigNumber, ethers } = require('ethers');
 const { WeiPerEther } = ethers.constants;
 
+const ASSET_ID_MAPPING = {
+  0: { symbol: 'ETH', decimals: 18 },
+  1: { symbol: 'DAI', decimals: 18 },
+  6: { symbol: 'USDC', decimals: 6 },
+};
```
Comment (rackstar) [inline src/lib/constants.js:8]: @MilGard91 I've added the mapping in the constants.js file for simplicity. should it be in the store file initialState?

File: src/store/reducer.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ const {
 
 const initialState = {
   assetRates: {}, // assetId -> rate
-  assets: { ETH: 0, DAI: 1, NXM: 255 },
+  assets: { ETH: 0, DAI: 1, USDC: 6, NXM: 255 },
```
Comment (rackstar) [inline src/store/reducer.js:11]: also is there a reason why we keyed  the assets off the assetSymbol? cover-router usually receives assetId. this line in selectAssetSymbol function could be made simpler if its keyed off assetId:

```
  return Object.keys(assets).find(key => assets[key] === assetId); // keyed off assetSymbol
  return assets[assetId]; // keyed off assetId
```

File: src/lib/constants.js (hunk 2 above)
Comment (MilGard91) [reply src/lib/constants.js]: Yes, sounds good if it not big of a refactor

Comment (MilGard91) [reply src/lib/constants.js]: Also add `NXM`

File: src/routes/quote.js (hunk 1 above)
Comment (MilGard91) [reply src/routes/quote.js]: We should keep it

File: src/store/reducer.js (hunk 3 above)
Comment (MilGard91) [reply src/store/reducer.js]: We use it for readability

**Review by MilGard91 — COMMENTED** [review]
We should also add decimals and symbol to capacity engine

File: src/lib/constants.js (hunk 2 above)
Comment (rackstar) [reply src/lib/constants.js]: I've added decimals in the end in the store so there is one source of truth for assets info

**Comment by rackstar** [comment]
@MilGard91 FE guys agreed that being consistent is better. so we went with:

```json
{
  "quote": {
    ...
    "asset": { <--- asset object containing all info
      "id": "string",
      "symbol": "string",
      "decimals": 18
    }
  },
  "capacities": [
    {
      "poolId": "string",
      "capacity": [
        {
          ...
          "asset": { <--- asset object containing all info
            "id": "string",
            "symbol": "string",
            "decimals": 18
          }
        }
      ]
    }
  ]
}
```

**Review by MilGard91 — COMMENTED** [review]
I would also bump the deployments package, everything else LGTM

**Comment by roxdanila** [comment]
I already bumped the deployments package, just rebase

**Comment by rackstar** [comment]
> I would also bump the deployments package, everything else LGTM

done

**Review by rackstar — APPROVED** [review]
approving as @MilGard91 says my asset field changes are good

**Pull request merged**